src/accelerate/big_modeling.py

```diff
@@ -379,6 +384,22 @@ def dispatch_model(
         )
         # Attaching the hook may break tied weights, so we retie them
         retie_parameters(model, tied_params)
+
+        # add warning to cuda and to method
+        def add_warning(fn):
+            @wraps(fn)
+            def wrapper(*args, **kwargs):
+                logger.warning("You can't move a model when it is dispatched on multiple devices.")
+                try:
+                    return fn(*args, **kwargs)
+                except NotImplementedError as e:
+                    raise RuntimeError("You can't move a model that has some modules offloaded to cpu or disk.") from e
```

Comment:
Maybe instead of the try except here we could detect if the model has some offloaded parameters and raise the error then.

Reply:
I do that by checking if any parameter is on `meta` device. Let me know that works for you. I don't check the device_map because people might remove the hooks after the dispatching the model.